x/finality/keeper/gov.go

```diff
@@ -94,27 +100,39 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 		)
 	}
 
-	// set the all the given finality providers voting power to 0
-	for h := uint64(haltingHeight); h <= uint64(currentHeight); h++ {
-		distCache := k.GetVotingPowerDistCache(ctx, h)
-		activeFps := distCache.GetActiveFinalityProviderSet()
-		for _, fpToJail := range fpPks {
-			fpDstInf, exists := activeFps[fpToJail.MarshalHex()]
-			if exists {
+	// set the all the given finality providers voting power to 0 and updates voting power distribution cache
+	for blkHeight := uint64(haltingHeight); blkHeight <= uint64(currentHeight); blkHeight++ {
+		dc := k.GetVotingPowerDistCache(ctx, blkHeight)
+		newDc := ftypes.NewVotingPowerDistCache()
+
+		for i := range dc.FinalityProviders {
+			// create a copy of the finality provider
+			fp := *dc.FinalityProviders[i]
+			fpBTCPKHex := fp.BtcPk.MarshalHex()
+
+			_, shouldJail := fpPksToJail[fpBTCPKHex]
+			if shouldJail {
 				// if the fp was already slashed at that height, keep as it was
 				// and do not update to jailed.
-				if !fpDstInf.IsSlashed {
-					fpDstInf.IsJailed = true
+				if !fp.IsSlashed {
+					fp.IsJailed = true
 				}
 
-				k.SetVotingPower(ctx, fpToJail.MustMarshal(), h, 0)
+				k.SetVotingPower(ctx, fp.BtcPk.MustMarshal(), blkHeight, 0)
 			}
+
+			// add this finality provider to the new cache if it has voting power
+			newDc.AddFinalityProviderDistInfo(&fp)
 		}
 
-		distCache.ApplyActiveFinalityProviders(params.MaxActiveFinalityProviders)
+		// sets the total vp in the new dc
+		newDc.ApplyActiveFinalityProviders(dc.NumActiveFps)
```

Comment:
why changing `params.MaxActiveFinalityProviders` to `dc.NumActiveFps` ? is this intentional to keep amount of active fp's at the same level as in old dc ?

Reply:
yes, as it might be different param at the block height of that vp distribution cache